subsys/dfu/boot/mcuboot_priv.h

```diff
@@ -10,6 +10,39 @@
 
 #include <storage/flash_map.h>
 
+#define DT_CODE_PARTITION_NODE DT_CHOSEN(zephyr_code_partition)
+/*
+ * If code partition is selected via chosen zephyr,code-partition it should
+ * be possible to automatically detect primary and secondary partition.
+ */
+#if DT_NODE_EXISTS(DT_CODE_PARTITION_NODE)
+#define DT_IMAGE_0_NODE DT_NODE_BY_FIXED_PARTITION_LABEL(image_0)
+#define DT_IMAGE_1_NODE DT_NODE_BY_FIXED_PARTITION_LABEL(image_1)
+#define DT_IMAGE_0_NS_NODE DT_NODE_BY_FIXED_PARTITION_LABEL(image_0_nonsecure)
```

Comment:
Will this fail for non-trustzone enabled devices where these `_nonsecure` partitions are not defined?

Reply:
@hakonfam Yes, probably yes. It assumes that they are there.

Reply:
> 
> 
> Will this fail for non-trustzone enabled devices where these `_nonsecure` partitions are not defined?

@hakonfam This shouldn't be the issue because this code only matches alternatives, so when `zephyr,code-partition` is `image_x` it will match the other `image_y` as DFU target; when the `zephyr,code-partition` is `image_x_nonsecure` it will try to match `image_y_nonsecure`.
If the alternative, the determined DFU partition does not exist than the DFU may no proceed anyway so compilation may and here.
The auto-code does not care about `CONFIG_TRUSTED*` definition as it works on what the `zephyr,code-partition` provides.
How is the `zephyr,code-partition` determined when `CONFIG_TRUSTED_EXECUTION_NONSECURE` is enabled?